tgz.go

```diff
@@ -20,12 +24,21 @@ type ArchiveWriter struct {
 	Writer  *bytes.Buffer
 }
 
+type ArchiveWriteTask struct {
+	Name    string
+	Content []byte
+}
+
 func (a *ArchiveWriter) Write(p []byte) (n int, err error) {
 	return a.Writer.Write(p)
 }
 
 func (a *ArchiveWriter) Close() error {
-	return a.Archive.AddFileByContent(a.Writer.Bytes(), a.File)
+	// create a write task from the data in this file
+	writeTask := ArchiveWriteTask{Name: a.File, Content: a.Writer.Bytes()}
+	// and queue it in the archive write queue
+	a.Archive.WriteQueue <- writeTask
```

Comment:
Suggestion:

```go
a.Archive.WriteQueue <- ArchiveWriteTask{
	Name: a.File,
	Content: a.Writer.Bytes(),
}
```

Now, the responsibilities seem a bit mixed, it is strange that an `ArchiveWriter` needs direct access to a channel/field in an `Archive`.